app/models/ems_folder.rb

```diff
@@ -195,4 +209,11 @@ def self.child_folder_paths_recursive(subtree, options = {})
     end
   end
   private_class_method :child_folder_paths_recursive
+
+  def self.folder_options(folders, options)
```

Comment:
Can you name this "apply_folder_options".  "folder_options" is a noun, so implies you are asking for some object.  "apply_" turns it into a verb making it clear than an action is being taken on the params.  You *may* want to make it "apply_folder_options!" to signify that it's modifying the params.